Frontend/Pages/SessionTimedOut.cshtml.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Configuration;
+
+namespace Frontend.Pages
+{
+    public class SessionTimedOut : PageModel
+    {
+        private readonly IConfiguration _configuration;
+
+        public SessionTimedOut(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+        public string ReturnUrl { get; set; }
```

Comment:
Not necessary but think you can add [FromQuery] attribute to this property and use model binding - then don't need to get it from Request.Query below

Reply:
oh yes, great point